File: src/lib/rcfile.js
const fs = require("fs");
const path = require("path");
const os = require("os");

const RCPATH = path.resolve(os.homedir(), ".comcordrc");

function readFile(config) {
  const rc = fs.readFileSync(RCPATH, "utf8");
  const lines = rc.split("\n");
  for (const line of lines) {
    const [key, value] = line.split("=");
    config[key] = value;
  }
}

function writeFile(config) {
  if (fs.existsSync(RCPATH)) {
    readFile(config);
  }
  const newrc = [];

  for (const key in config) {
    const value = config[key];
    newrc.push(`${key}=${value}`);
  }

  fs.writeFileSync(RCPATH, newrc.join("\n"));
}

module.exports = {readFile, writeFile, path: RCPATH};

File: rcfile.go
package main

import (
  "os"
  "path/filepath"
  "strings"
)

func GetRCPath() string {
  homeDir, err := os.UserHomeDir()
  if err != nil {
    panic(err)
  }

  return filepath.Join(homeDir, ".comcordrc")
}

func LoadRCFile() map[string]string {
  config := make(map[string]string)
  file, err := os.ReadFile(GetRCPath())
  if err != nil {
    panic(err)
  }

  lines := strings.Split(string(file), "\n")
  for _, line := range lines {
    kvs := strings.Split(line, "=")
    if len(kvs) == 2 {
      config[kvs[0]] = kvs[1]
    }
  }

  return config
}

func SaveRCFile(config map[string]string) {
  out := ""

  for key, value := range config {
    out = out + key + "=" + value + "\n"
  }

  err := os.WriteFile(GetRCPath(), []byte(out), 0644)
  if err != nil {
    panic(err)
  }
}
